### PR TITLE
Problem: Multiple fd_t definitions

### PR DESCRIPTION
**Pull request by gummif**
Solution: Unify definition

### CODE REVIEW

File: tests/testutil.hpp [hunk 1]
```diff
@@ -37,6 +37,7 @@
 #endif
 #include "../include/zmq.h"
 #include "../src/stdint.hpp"
+#include "../src/fd.hpp"
```
Comment (bluca) [inline tests/testutil.hpp:40]: Please don't - I do not want to start including headers from the tests. Yes there's already one or two - they should be removed, not the other way around.

Comment (gummif) [reply tests/testutil.hpp]: Fixed

File: tests/test_poller.cpp [hunk 2]
```diff
@@ -41,9 +41,9 @@
 
 SETUP_TEARDOWN_TESTCONTEXT
 
-fd_t get_fd (void *socket_)
+zmq_fd_t get_fd (void *socket_)
```
Comment (bluca) [inline tests/test_poller.cpp:44]: why the renaming churn? it's just tests, there's no need to namespace internal types

Comment (gummif) [reply tests/test_poller.cpp]: fixed

**Pull request merged**